ga4gh/datamodel/datasets.py

```diff
@@ -227,6 +226,27 @@ def __init__(self, localId, dataDir, backend):
                     self, localId, bamPath, backend)
                 self.addReadGroupSet(readGroupSet)
 
+    def descriptionExists(self, fname):
+        # check for existence of description JSON
+        return os.path.exists(fname)
+
+    def parseDescriptionJson(self, str):
+        # parse into python dict
+        return json.loads(str)
+
+    def descriptionFilename(self, path, localId):
+        return os.path.join(path, localId + ".json")
+
+    def isAnnotated(self, path, localId):
+        ret = False
+        fname = self.descriptionFilename(path, localId)
+        if self.descriptionExists(fname):
```

Comment:
`Description' is confusing terminology.  It's generally though
of as natural language text.  Metadata would be much clearer
name.  Please change.


Jerome Kelleher <notifications@github.com> writes:
> In ga4gh/datamodel/datasets.py:
> 
> > @@ -227,6 +226,27 @@ def __init__(self, localId, dataDir, backend):
> >                      self, localId, bamPath, backend)
> >                  self.addReadGroupSet(readGroupSet)
> >
> > +    def descriptionExists(self, fname):
> > +        # check for existence of description JSON
> > +        return os.path.exists(fname)
> > +
> > +    def parseDescriptionJson(self, str):
> > +        # parse into python dict
> > +        return json.loads(str)
> > +
> > +    def descriptionFilename(self, path, localId):
> > +        return os.path.join(path, localId + ".json")
> > +
> > +    def isAnnotated(self, path, localId):
> > +        ret = False
> > +        fname = self.descriptionFilename(path, localId)
> > +        if self.descriptionExists(fname):
> 
> I'm not sure the descriptionExists does much for readablility here.How about
> 
> fname = self.getDescriptionFilename(path, localId)
> if os.path.exists(fname):
>     # etc.
> 
> —
> Reply to this email directly or view it on GitHub.*
>